### PR TITLE
fix(onboarding): ensures proper returnTo after email validation

### PR DESCRIPTION
**Pull request by ygrishajev**
## Why

Users were stuck in a redirect loop after email verification. `OnboardingRedirectEffect` redirected users from `/user/verify-email` to onboarding with `returnTo` set to `/user/verify-email?email=...`. After completing onboarding, users were sent back to verify-email, which triggered the onboarding redirect again — creating an infinite loop. The root cause was that `/user/verify-email` was not in the excluded paths list for the onboarding redirect.

## What

- Added `/user/verify-email` to `EXCLUDED_PREFIXES` in `OnboardingRedirectEffect` so the verify-email page is not intercepted by the onboarding redirect
- Added `{ returnTo: "/" }` to the `UrlService.onboarding()` call in the verify-email page so users navigate to `/` after completing onboarding
- Extracted `VerifyEmailPage` from the Next.js page file into its own component under `components/onboarding/VerifyEmailPage/`
- Added dependency injection to `OnboardingRedirectEffect` and `VerifyEmailPage` for testability
- Added unit tests for `OnboardingRedirectEffect` (11 tests) and `VerifyEmailPage` (6 tests)

## Summary by CodeRabbit

* **New Features**
  * Added a dedicated email verification page to support onboarding email verification.
  * Updated redirect behavior to exclude the verify-email path and handle return-to encoding.

* **Tests**
  * Added unit tests covering email verification flows: success, error, loading, and missing params.
  * Added tests for onboarding redirect logic across user/wallet/loading edge cases.

### CODE REVIEW

File: apps/deploy-web/src/components/onboarding/OnboardingRedirectEffect/OnboardingRedirectEffect.tsx [hunk 1]
```diff
@@ -5,12 +5,23 @@ import { useWallet } from "@src/context/WalletProvider";
...
 
-export const OnboardingRedirectEffect = () => {
-  const { user, isLoading: isUserLoading } = useUser();
-  const { hasManagedWallet, isWalletConnected, isWalletLoading } = useWallet();
-  const router = useRouter();
+const DEPENDENCIES = {
+  useUser,
+  useWallet,
+  useRouter,
+  UrlService
```
Comment (stalniy) [inline apps/deploy-web/src/components/onboarding/OnboardingRedirectEffect/OnboardingRedirectEffect.tsx:14]: **suggestion(non-blocking):** `urlService` is exposed via `useServices` hook

**Review by stalniy — APPROVED** [review]
